### PR TITLE
Fix missing llm in Inline Edit

### PR DESCRIPTION
**Pull request by mkondratek**
This PR is a reattempt of https://github.com/sourcegraph/jetbrains/pull/2084. 

API changed in Cody. `model` becomes `id`. 

## Test plan
1. Edit Code works

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt [hunk 1]
```diff
@@ -151,5 +153,21 @@ class HistoryService(private val project: Project) :
 
   companion object {
     @JvmStatic fun getInstance(project: Project): HistoryService = project.service<HistoryService>()
+
+    fun convertModelsToChatModelProviders(models: List<Model>) =
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt]: `Model` and `ChatModelProvider` are in fact the same data structure, one autogenerated and one crafted by us, no?
Why not to replace it completely?

Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt]: done, ~however we need to leave the class for the sake of some old migrations that could have the separate deprecated and isCodyProOnly fields in their state~ 

nah, we can remove it - we did not persist old fashioned field, we have the tags only

**Review by pkukielka — APPROVED** [review]
Thank you! LGTM now

**Comment by pkukielka** [comment]
```
> Task :test
CompileCommand: exclude com/intellij/openapi/vfs/impl/FilePartNodeRoot.trieDescend bool exclude = true

SettingsMigrationTest > test ChatTagsLlmMigration FAILED
    junit.framework.AssertionFailedError at SettingsMigrationTest.kt:351
```

**Pull request merged**